flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapManager.java

```diff
@@ -193,11 +222,78 @@ private TargetNode getTargetNode(Node parentNode, String targetField, SemanticMo
             return new TargetNode(typeSymbol, variableSymbol.getName().get(), null);
         }
         ExpressionNode initializer = optInitializer.get();
+        if (initializer.kind() == SyntaxKind.FUNCTION_CALL && nodeKind == NodeKind.FUNCTION_CALL) {
+            FunctionCallExpressionNode funcCallExprNode = (FunctionCallExpressionNode) initializer;
+            Optional<Symbol> optFunctionSymbol = semanticModel.symbol(funcCallExprNode);
+            if (optFunctionSymbol.isEmpty() || optFunctionSymbol.get().kind() != SymbolKind.FUNCTION) {
+                return null;
+            }
+            FunctionSymbol functionSymbol = (FunctionSymbol) optFunctionSymbol.get();
+            Optional<List<ParameterSymbol>> optParams = functionSymbol.typeDescriptor().params();
+            if (optParams.isEmpty()) {
+                return null;
+            }
+            List<ParameterSymbol> paramSymbols = optParams.get();
+            for (int i = 0; i < paramSymbols.size(); i++) {
+                ParameterSymbol paramSymbol = paramSymbols.get(i);
+                if (paramSymbol.paramKind() != ParameterKind.REQUIRED) {
+                    continue;
+                }
+                if (paramSymbol.getName().get().equals(propertyKey)) {
+                    FunctionArgumentNode arg = funcCallExprNode.arguments().get(i);
+                    if (arg.kind() == SyntaxKind.POSITIONAL_ARG) {
+                        return new TargetNode(paramSymbol.typeDescriptor(), propertyKey,
+                                ((PositionalArgumentNode) arg).expression());
+                    }
+                    break;
+                }
+            }
+            return null;
+        } else if (initializer.kind() == SyntaxKind.CHECK_EXPRESSION && nodeKind == NodeKind.NEW_CONNECTION) {
+            ExpressionNode expressionNode = ((CheckExpressionNode) initializer).expression();
+            if (expressionNode.kind() != SyntaxKind.IMPLICIT_NEW_EXPRESSION) {
+                return null;
+            }
+            TypeSymbol rawType = CommonUtils.getRawType(typeSymbol);
+            if (rawType.kind() != SymbolKind.CLASS) {
+                return null;
+            }
+            ClassSymbol classSymbol = (ClassSymbol) rawType;
+            Optional<MethodSymbol> optInitMethodSymbol = classSymbol.initMethod();
+            if (optInitMethodSymbol.isEmpty()) {
+                return null;
+            }
+            MethodSymbol initMethodSymbol = optInitMethodSymbol.get();
+            Optional<List<ParameterSymbol>> optParams = initMethodSymbol.typeDescriptor().params();
+            if (optParams.isEmpty()) {
+                return null;
+            }
+            ImplicitNewExpressionNode implicitNewExprNode = (ImplicitNewExpressionNode) expressionNode;
+            List<ParameterSymbol> paramSymbols = optParams.get();
+            for (int i = 0; i < paramSymbols.size(); i++) {
+                ParameterSymbol paramSymbol = paramSymbols.get(i);
```

Comment:
Can we extract this logic to a separate method and reuse for Line 240-252 as well?

Reply:
FIxed